### PR TITLE
PLFM-1319: support new registration token, and change pom to 2012-07-25-e2adb1e-22 (services-repository artifact that contain the user registration change)

### PR DESCRIPTION
**Pull request by jay-hodgson**
Also has the changes for SWC-137 (issues brought up during a previous code review).

### CODE REVIEW

File: src/main/java/org/sagebionetworks/web/client/presenter/users/PasswordResetPresenter.java [hunk 1]
```diff
@@ -64,6 +66,11 @@ public void setPlace(PasswordReset place) {
 		// show proper view if token is present
 		if(DisplayUtils.DEFAULT_PLACE_TOKEN.equals(place.toToken())) {
 			view.showRequestForm();
+		} else if (place.toToken().startsWith("register_")) {
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/presenter/users/PasswordResetPresenter.java]: make this a constant

File: src/main/java/org/sagebionetworks/web/server/servlet/SynapseClientImpl.java [hunk 2]
```diff
@@ -619,11 +623,39 @@ public void updateUserProfile(String userProfileJson)
...
+			UserProfile profile = EntityFactory.createEntityFromJSONObject(userProfileJSONObject, UserProfile.class);
+			AttachmentData pic = profile.getPic();
+			if (pic != null && pic.getTokenId() == null && pic.getUrl() != null){
+				//special case, client provided just enough information to pull the pic from an external location (so store in s3 before updating the profile).
+				log.info("Downloading picture from url: " + pic.getUrl());
+				URL url = new URL(pic.getUrl());
+			    URLConnection conn = url.openConnection();
+			    conn.setDoInput(true);
+			    conn.setDoOutput(false);
+			    File temp = ServiceUtils.writeToTempFile(conn.getInputStream(), UserProfileAttachmentServlet.MAX_ATTACHMENT_SIZE_IN_BYTES);
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/server/servlet/SynapseClientImpl.java:641]: in finally, close conn.getInputStream()

File: src/main/java/org/sagebionetworks/web/server/servlet/UserAccountServiceImpl.java [hunk 3]
```diff
@@ -283,7 +328,17 @@ public String initiateSession(String username, String password, boolean explicit
...
 				String sessionToken = initSession.getSessionToken();
 				UserProfile profile = getUserProfile(sessionToken);
-				profile.setUserName(username);
+				String fName = null;
+				String lName = null;
+				if (initSession.getDisplayName() != null) {
+					String[] firstLast = initSession.getDisplayName().split(" ");
+					if (firstLast.length == 2) {
+						fName = firstLast[0];
+						lName = firstLast[1];
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/server/servlet/UserAccountServiceImpl.java]: lName should be the rest of the string (not the second token)

File: src/main/java/org/sagebionetworks/web/client/presenter/users/PasswordResetPresenter.java [hunk 4]
```diff
@@ -107,29 +114,38 @@ public void onFailure(Throwable caught) {
 
 	@Override
 	public void resetPassword(final String newPassword) {
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/presenter/users/PasswordResetPresenter.java:117]: Please test this method

File: src/test/java/org/sagebionetworks/web/unitclient/widget/breadcrumb/BreadcrumbTest.java [hunk 5]
```diff
@@ -132,6 +136,25 @@ public void testAsWidgetEntity() throws Exception {
...
+		links.add(homeLink);
+		String currentPageName  = "CurrentPage";
+		breadcrumb.asWidget(links, currentPageName);
+		verify(mockView).setLinksList(links, currentPageName);
+		
+		//also verify the single page off of home works
+		reset(mockView);
+		breadcrumb.asWidget(currentPageName);
+		verify(mockView).setLinksList(any(List.class), (String)isNotNull());
+	}
```
Comment (jay-hodgson) [inline src/test/java/org/sagebionetworks/web/unitclient/widget/breadcrumb/BreadcrumbTest.java:155]: verify links are the links passed in. (may be able to replace any(List.class) with links.

**Comment by dburdick** [comment]
Awesome, thanks for adding the test.

**Pull request merged**